packages/core/src/registry/registry.ts

```diff
@@ -0,0 +1,69 @@
+import { RouteHandlerMethod } from "fastify";
+import { RESTMethods } from "../loader/rest";
+
+export type RegistryMock = {
+  method: RESTMethods;
+  path: string;
+  resolver: RouteHandlerMethod;
+  isDefault: boolean;
+  name: string;
+};
+
+export default class Registry {
+  private activeMocks: Map<string, RegistryMock> = new Map();
+
+  private allMocks: RegistryMock[] = [];
+
+  public loadAllMocks = (mocks: RegistryMock[]) => {
+    mocks.forEach((mock) => {
+      this.allMocks.push(mock);
+    });
+  };
+
+  public loadDefaultMocks = (mocks: RegistryMock[]) => {
+    mocks
+      .filter((mock: RegistryMock) => mock.isDefault)
+      .forEach((mock) => {
+        this.activeMocks.set(`${mock.method} ${mock.path}`, mock);
```

Comment:
I have one alternative design in mind, where we skip the notion of default mock, method and path and simply do one huge pipe and defer resolution mechanism to the client. In other words, all handlers would be:

```
app.all(handler)
...
const handler = (req, res) => {
 // I'll be executed every time;
 if (req.method === 'POST' && req.path == ...) // I decide here
} 

// we can give people utils like this:

const handler = postHandler((req, res) => {
...
})

const handler = postHandler(pathHandler('/talent', (req, res) => {
...
}))
```

Simplifies a lot in this class. Up to you. This is fine what we have, but we're stuck with isDefault, path and method.

Reply:
I would merge with the current solution, but I expect API to change in the future. We can adapt to our / users' needs. 
Let's remember this comment 👍